ios/PresentationLayer/UIToolkit/Sources/UIToolkit/BaseViews/SwiftUI/Image/RemoteImage.swift

```diff
@@ -0,0 +1,91 @@
+//
+//  Created by David Kadlček on 12.05.2023
+//  Copyright © 2023 Matee. All rights reserved.
+//
+
+import Foundation
+import SwiftUI
+
+public struct RemoteImage: View {
+    
+    @State private var image: Image?
+    private let stringURL: String?
+    private let defaultImage: Image
+    private let contentMode: ContentMode
+    private let failureMessage: String?
+    
+    public init(
+        stringURL: String?,
+        defaultImage: Image,
+        contentMode: ContentMode = .fit,
+        failureMessage: String? = nil
+    ) {
+        self.stringURL = stringURL
+        self.defaultImage = defaultImage
+        self.contentMode = contentMode
+        self.failureMessage = failureMessage
+    }
+    
+    public var body: some View {
+        if let image {
+            image
+                .resizable()
+                .aspectRatio(contentMode: contentMode)
+        } else if let stringURL = stringURL, let url = URL(string: stringURL) {
+            placeholderContent()
+                .skeleton(true)
+                .onAppear(perform: { downloadRemoteImage(from: url) })
+        } else {
+            placeholderContent()
+        }
+    }
+}
+    
+private extension RemoteImage {
+    
+    // MARK: - Subviews
+
+    @ViewBuilder
+    func placeholderContent() -> some View {
+        defaultImage
+            .resizable()
+            .aspectRatio(contentMode: contentMode)
+    }
+
+    @ViewBuilder
+    func failureContent() -> some View {
+        if let failureMessage = failureMessage {
+            Text(failureMessage)
+                .font(.caption)
+                .fontWeight(.medium)
+        } else {
+            placeholderContent()
+        }
+    }
+    
+    // MARK: - Remote image downloading
+    
+    func downloadRemoteImage(from url: URL) {
+        Task {
+            self.image = await downloadImage(from: url)
+        }
+    }
+    
+    func downloadImage(from url: URL) async -> Image? {
+        do {
+            let cache = URLCache.shared
+            let urlRequest = URLRequest(url: url)
+            
+            let (data, response) = try await URLSession.shared.data(for: urlRequest)
+            
+            if cache.cachedResponse(for: urlRequest) == nil {
+                cache.storeCachedResponse(CachedURLResponse(response: response, data: data), for: urlRequest)
+            }
```

Comment:
There must be a reason why the built-in caching doesn't work.. We use it in Můj Up and it works as expected. 

Also, doesn't this entirely block you from working with the cache correctly? Doesn't this just cache the response for a specific URL and then never update it? What if you want to force using only a remote resource? Or what if the cached response exceeds its max age?

There's probably a lot of logic behind the cache policy, which is often specified by the server in the `Cache-Control` header, and `URLCache` has all the logic implemented for handling that correctly.

It just seems to be a bit unusable in many scenarios when such cases need to be handled correctly, which may not make the component universal enough.

Reply:
It is really weird. I tried everything and nothing works... So we can approve this MR and create some task to improve it 🥇